File: src/2520-Count-the-Digits-That-Divide-a-Number/count_the_digits_that_divide_a_number.cpp
#include <iostream>
#include <string>

class Solution {
public:
    int countDigits(int num) {
        std::string s = std::to_string(num);
        int ans = 0;

        for (char n : s) {
            if (num % (n - '0') == 0) {
                ans += 1;
            }
        }

        return ans;
    }
};

int main() {
    int num = 121;
    Solution solution;

    std::cout << solution.countDigits(num);

    return 0;
}

File: src/2520-Count-the-Digits-That-Divide-a-Number/count_the_digits_that_divide_a_number.py
class Solution:
    def countDigits(self, num: int) -> int:
        s: str = str(num)
        ans: int = 0
        
        for n in s:
            
            if num  % int(n) == 0:
                ans += 1

        return ans
    
if __name__ == "__main__":
    num = 121
    solution = Solution()
    
    print(solution.countDigits(num))